### ISSUE TITLE
language: support function call on for loops

### ISSUE THREAD

**Issue by katcipis** [NeowayLabs/nash#158]
Example:

```
 for path in split($paths, "\n") {
                mv $srcdir + $path $vendor
        }
```

Should work, today it explodes on your face

**Comment by katcipis** [comment]
It explodes nicely with a ``` Expected variable but found  IDENT``` error :-)

**Comment by katcipis** [comment]
This would be great too:

```
 for path in split(<= somecmd, "\n") {
                mv $srcdir + $path $vendor
 }
```

**Comment by tiago4orion** [comment]
> split(<= somecmd, "\n")

Interesting. I had thought a lot about this and didn't got this idea. Thanks!

**Comment by katcipis** [comment]
It is inspired on the Go channel :-). I love the idea of Nash being like a Go with shell super powers. And without concurrency, since it makes things extremely messy and are not necessary for simple scripting :-)

**Comment by tiago4orion** [comment]
Closed by #168
